awswrangler/_utils.py

```diff
@@ -83,24 +84,31 @@ def _get_endpoint_url(service_name: str) -> Optional[str]:
     return endpoint_url
 
 
+@apply_configs
 def client(
-    service_name: str, session: Optional[boto3.Session] = None, config: Optional[botocore.config.Config] = None
+    service_name: str, session: Optional[boto3.Session] = None, botocore_config: Optional[botocore.config.Config] = None
```

Comment:
I think keeping the parameter as `config` would make more sense here, to keep a 1:1 mapping to param names for `boto3.Session.client()`.

Reply:
The problem is that with the way that the `awswrangler.config` is implemented is that the function parameter needs to have the same name as the config parameter. This would mean that you would have to set `awswrangler.config.config` for changing the `botocore`, which can be misleading in my opinion.